sysvabi64/sysvabi64.rst

```diff
@@ -0,0 +1,1076 @@
+    ret
+    .align 3
+  .LC0:
+    .xword  weakref                      // R_AARCH64_ABS64 weakref
+
+  .bss
+  .LANCHOR0
+  ldst:
+    .zero   1
+    .zero   7
+  lbdst:
+    .zero   65536
+
+Large Code Model
+^^^^^^^^^^^^^^^^
+
+The large model makes no assumptions about the size of the static code
+and data segments, and therefore uses absolute 64-bit addresses in all
+cases. As well as massive GNU/Linux executables, the large code model
+would also be suitable for bare-metal programs where a linker script
+causes sections to be mapped sparsely within the virtual address
+space, or where a high-level code references absolute symbols defined
+in a linker script.
+
+.. code-block:: asm
+
+  foo:
+    adrp    x0, .LC0              // R_AARCH64_ADR_PREL_PG_HI21   .LC0
+    ldr     x0, [x0, #:lo12:.LC0] // R_AARCH64_LDST64_ABS_LO12_NC .LC0
+    adrp    x1, .LC1              // R_AARCH64_ADR_PREL_PG_HI21   .LC1
+    ldr     x1, [x1, #:lo12:.LC1] // R_AARCH64_LDST64_ABS_LO12_NC .LC1
+    ldrb    w1, [x1]
+    strb    w1, [x0]
+    adrp    x1, .LC2              // R_AARCH64_ADR_PREL_PG_HI21   .LC2
+    ldr     x1, [x1, #:lo12:.LC2] // R_AARCH64_LDST64_ABS_LO12_NC .LC2
+    str     x0, [x1]
+    ret
+    .align 3
+  .LC0:
+    .xword  dst                   // R_AARCH64_ABS64 .LC0
+  .LC1:
+    .xword  src                   // R_AARCH64_ABS64 .LC1
+  .LC2:
+    .xword  ptr                   // R_AARCH64_ABS64 .LC2
+
+  bar:
+    adrp    x0, .LC3              // R_AARCH64_ADR_PREL_PG_HI21 .LC3
+    ldr     x0, [x0, #:lo12:.LC3] // R_AARCH64_LDST64_ABS_LO12_NC .LC3
+    ldrb    w1, [x0, 1]
+    strb    w1, [x0]
+    str     x0, [x0, 8]
+    ret
+    .align  3
+  .LC3:
+    .xword  .LANCHOR0             // R_AARCH64_ABS64 .LANCHOR0
+
+  baz:
+    adrp    x0, .LC4              // R_AARCH64_ADR_PREL_PG_HI21 .LC4
+    ldr     x0, [x0, #:lo12:.LC4] // R_AARCH64_LDST64_ABS_LO12_NC .LC4
+    adrp    x1, .LC5              // R_AARCH64_ADR_PREL_PG_HI21 .LC5
+    ldr     x1, [x1, #:lo12:.LC5] // R_AARCH64_LDST64_ABS_LO12_NC .LC5
+    ldrb    w1, [x1]
+    strb    w1, [x0]
+    adrp    x1, .LC6              // R_AARCH64_ADR_PREL_PG_HI21 .LC6
+    ldr     x1, [x1, #:lo12:.LC6] // R_AARCH64_LDST64_ABS_LO12_NC .LC6
+    str     x0, [x1, 8]
+    ret
+    .align  3
+  .LC4:
+    .xword  lbdst                 // R_AARCH64_ABS64 lbdst
+  .LC5:
+    .xword  lbsrc                 // R_AARCH64_ABS64 lbsrc
+  .LC6:
+    .xword  .LANCHOR0             // R_AARCH64_ABS64 .LANCHOR0
+
+  getweakaddr:
+    adrp    x0, .LC7              // R_AARCH64_ADR_PREL_PG_HI21 .LC7
+    ldr     x0, [x0, #:lo12:.LC7] // R_AARCH64_LDST64_ABS_LO12_NC .LC7
+    ret
+  .LC7:
+    .align 3
+    .xword  weakref               // R_AARCH64_ABS64 weakref
+
+  .bss
+  .LANCHOR0
+  ldst:
+    .zero   1
+    .zero   7
+  lbdst:
+    .zero   65536
+
+
+Position independent addressing
+-------------------------------
+
+In position independent code (PIC) the virtual addresses of
```

Comment:
These paragraphs explaining GOT should go much earlier (before we list GOT relocations and GOT limits).

Reply:
OK done